File: app/src/main/java/com/kakacat/minitool/common/util/ThreadUtil.java
package com.kakacat.minitool.common.util;

import java.util.concurrent.Callable;

import bolts.Task;

public class ThreadUtil {

    public static void callInUiThread(Runnable runnable) {
        Task.call(runToCall(runnable), Task.UI_THREAD_EXECUTOR);
    }

    public static void callInBackground(Runnable runnable) {
        Task.callInBackground(runToCall(runnable));
    }

    private static Callable<Void> runToCall(Runnable runnable) {
        return () -> {
            runnable.run();
            return null;
        };
    }
}

File: app/src/main/java/com/kakacat/minitool/common/util/ThreadUtil.kt
package com.kakacat.minitool.common.util

import bolts.Task
import java.util.concurrent.Callable

object ThreadUtil {
    @JvmStatic
    fun callInUiThread(runnable: Runnable) {
        Task.call(runToCall(runnable), Task.UI_THREAD_EXECUTOR)
    }

    @JvmStatic
    fun callInBackground(runnable: Runnable) {
        Task.callInBackground(runToCall(runnable))
    }

    private fun runToCall(runnable: Runnable): Callable<Void?> {
        return Callable {
            runnable.run()
            null
        }
    }
}